sno/crs_util.py

```diff
@@ -1,14 +1,32 @@
 from osgeo.osr import SpatialReference
 
 
-def get_identifier(crs):
+def get_identifier_str(crs):
     """
-    Given a CRS, generate a unique idenfier for it. Eg: "EPSG:2193"
+    Given a CRS, generate a stable, unique identifier for it of type 'str'. Eg: "EPSG:2193"
     """
     if isinstance(crs, str):
         crs = SpatialReference(crs)
-    if isinstance(crs, SpatialReference):
-        auth_name = crs.GetAuthorityName(None)
-        auth_code = crs.GetAuthorityCode(None)
+    if not isinstance(crs, SpatialReference):
+        raise RuntimeError(f"Unrecognised CRS: {crs}")
+    auth_name = crs.GetAuthorityName(None)
+    auth_code = crs.GetAuthorityCode(None)
+    if auth_name and auth_code:
         return f"{auth_name}:{auth_code}"
-    raise RuntimeError(f"Unrecognised CRS: {crs}")
+    return auth_name or auth_code
```

Comment:
Done - but only if the authority is useless like "0" or "EPSG" - a user could well have written a semi-descriptive unique authority and we in general respect that whatever they have called it, while it may not be the best name, might be better than an opaque hash.
AUTHORITY["", "PA27"]
or
AUTHORITY["olsen232", ""]